webjars-based foreign lib failure - Linux only

This project should reliably demonstrate the issue:

https://github.com/neilmock/figwheel-webjars

A gist demonstrating success on OSX and failure on Linux.  No ~/.lein/profiles are present on either setup.

https://gist.github.com/neilmock/ff066966aa75538412a9e6a9a8421c27

On Linux, "lein cljsbuild once" successfully compiles, so I assume this is an issue with the file watching mechanism of figwheel (or hawk) not picking up on the classpath asset.  But, I'm unfortunately not able to isolate any further what the issue might be.

Thanks for the great tool!

Hi @bhauman any progress on this? We have also stepped on this one. In our case, this happens when using https://...googlemaps... in our foreign-libs in our project.clj. It does compile OK under OSX.

```
Caused by: java.nio.file.NoSuchFileException: /tmp/website/https:/maps.googleapis.com/maps/api
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
	at sun.nio.fs.UnixException.asIOException(UnixException.java:111)
	at sun.nio.fs.LinuxWatchService$Poller.implRegister(LinuxWatchService.java:246)
	at sun.nio.fs.AbstractPoller.processRequests(AbstractPoller.java:260)
	at sun.nio.fs.LinuxWatchService$Poller.run(LinuxWatchService.java:329)
	at java.lang.Thread.run(Thread.java:745)
```

--- a/sidecar/src/figwheel_sidecar/watching.clj
+++ b/sidecar/src/figwheel_sidecar/watching.clj
@@ -13,7 +13,8 @@
 
 (defn files-and-dirs [source-paths]
   (group-by #(if (.isDirectory %) :dirs :files)
-            (map io/file source-paths)))
+            (->> (map io/file source-paths)
+                 (filter #(.exists %)))))
 
 ;; relies on canonical strings
 (defn is-subdirectory [dir child]

